### DISCUSSION TITLE
Can we just use the .js extension instead of .jsx?

### DISCUSSION THREAD

**Discussion by mjr** [remix-run/remix#1208] [Q&A] [answered]
I tried to use it, but I had some problems and I was unsure if the remix allows us to do this:

`app/components/button.js`
`app/routes/index.js`
`app/entry.server.js`
`app/entry.client.js`
`app/root.js`

**Comment by Girish21** [comment]
No, I guess, since Remix uses [`esbuild`](https://esbuild.github.io) as the compiler. And, `esbuild` requires the file extension to be [`jsx` or `tsx`](https://esbuild.github.io/content-types/#jsx)

**Reply by mjr** [reply]
Oh ok, thanks.

**Comment by mjackson** [comment, marked as answer]
Yes, you can use either .js or .jsx (or .ts or .tsx if you're using TypeScript). They all should be supported.

**Reply by mjr** [reply]
So can you help understand this problem i'm having?

Basically what I did was create a project using `npx create-remix@latest` and then rename the files `root.jsx`, `entry.client.jsx`, `entry.server.jsx` and `routes/index.jsx` to `.js`

<img width="838" alt="Screen Shot 2021-12-24 at 19 31 54" src="https://user-images.githubusercontent.com/9324986/147373510-4fbf77cb-7d17-44f7-a779-dd3aa41275cb.png">

https://codesandbox.io/s/goofy-kapitsa-8v7en

**Reply by pazguille** [reply]
Hi @mjr, @mjackson 

The problem is a missing configuration of `loader` in [routes.ts file](https://github.com/remix-run/remix/blob/main/packages/remix-dev/compiler/routes.ts#L48).

If I add the loaders works fine.

```js
  let result = await esbuild__namespace.build({
    entryPoints: [path__namespace.resolve(config.appDirectory, config.routes[routeId].file)],
    platform: "neutral",
    format: "esm",
    metafile: true,
    write: false,
    logLevel: "silent",
    loader: { '.js': 'jsx' }, // Here!
    plugins: [mdx.mdxPlugin(config)]
  });
```
